src/components/SettingsModal/SettingsModalComponent.js

```diff
@@ -318,28 +327,46 @@ const SettingsModalComponent = ({
                   return false;
                 })()}
                 onClick={() => {
+                  i18n.changeLanguage(selectedLocale.value);
```

Comment:
This could be moved to the redux dispatch for `updateUserLocale()` given it's intended responsibility.

Reply:
Moved method call to `src/redux/users/sagas.js`